Sources/Stevenson/SlackService.swift

```diff
@@ -11,18 +11,48 @@ public struct SlackCommand {
     /// If empty the command will be allowed in all channels
     public let allowedChannels: Set<String>
 
-    let run: (SlackCommandMetadata, Request) throws -> Future<SlackResponse>
+    /// Closure that performs the actual action of the command.
+    /// If subCommands are provided then it first will try to select approapraite sub command
+    /// by the first word in the command text, and if it finds one then this command will be executed,
+    /// otherwise this closure is called
+    public let run: (SlackCommandMetadata, Request) throws -> Future<SlackResponse>
 
     public init(
         name: String,
         help: String,
         allowedChannels: Set<String>,
+        subCommands: [SlackCommand] = [],
         run: @escaping (SlackCommandMetadata, Request) throws -> Future<SlackResponse>
     ) {
         self.name = name
         self.allowedChannels = allowedChannels
-        self.help = help
-        self.run = run
+        if subCommands.isEmpty {
+            self.help = help
+        } else {
+            self.help = help +
+            """
+            Sub-commands:
+            \(subCommands.map({ "- \($0.name)" }).joined(separator: "\n"))
+            """
```

Comment:
Nit: should we also add a hint "use `/stevenson <subcommand> help` for more info on a subcommand"? (As some people might try `/stevenson help <subcommand>` instead, which won't work)

Reply:
no